File: .github/scripts/delete-namespace.ts
import { formatNamespace, setupTurbopuffer } from "../utils.ts";

const { REPO_NAME, TURBOPUFFER_API_KEY } = Deno.env.toObject();

if (!REPO_NAME) throw new Error("REPO_NAME is not set");
if (!TURBOPUFFER_API_KEY) throw new Error("TURBOPUFFER_API_KEY is not set");

const turbopuffer = setupTurbopuffer();

const ns = turbopuffer.namespace(formatNamespace(REPO_NAME));
await ns.deleteAll();

console.log(`"${formatNamespace(REPO_NAME)}" deleted`);

File: .github/scripts/delete-namespace.py
from os import getenv

import turbopuffer as tp

REPO_NAME = getenv("REPO_NAME")
TURBOPUFFER_API_KEY = getenv("TURBOPUFFER_API_KEY")

if not REPO_NAME:
    raise ValueError("REPO_NAME is not set")

if not TURBOPUFFER_API_KEY:
    raise ValueError("TURBOPUFFER_API_KEY is not set")

tp.api_key = getenv("TURBOPUFFER_API_KEY")
tp.api_base_url = "https://gcp-us-east4.turbopuffer.com"

ns_code = tp.Namespace(f'site-{REPO_NAME}-code')
ns_blocks = tp.Namespace(f'site-{REPO_NAME}-blocks')

if not ns_code.exists() or not ns_blocks.exists():
    print(f"Namespace {REPO_NAME} does not exist")
    exit(0)

ns_code.delete_all()
ns_blocks.delete_all()
